### PR TITLE
Enable IdleTuning feature on zLinux and pLinux

### PR DESCRIPTION
**Pull request by Param-S**
Enables the idle tuning feature for zLinux and pLinux which is currently disabled.

Refer https://github.com/eclipse/omr/pull/1069 for more details on IdleTuning.

Signed-off-by: Parameswaran Selvam <parselva@in.ibm.com>

### CODE REVIEW

**Comment by DanHeidinga** [comment]
> Refer eclipse/omr#1069 for more details on IdleTuning.

Which says: 
> This PR is currently scoped to only Generational Concurrent GC policy and Linux
x86(32bit + 64bit) platform.

@Param-S Is there another OMR PR required to add support for non-x86 platforms before merging this?

**Comment by Param-S** [comment]
It does not require OMR changes. We only need to enable the flags in OpenJ9.

**Comment by DanHeidinga** [comment]
Thanks for confirming @Param-S - please remove the "WIP" from the title when you're ready to have this reviewed

**Comment by Param-S** [comment]
@pshipton Please review the changes.

File: runtime/vm/jvminit.c [hunk 1]
```diff
@@ -1952,7 +1952,7 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 			 */
 			if (J9_ARE_ANY_BITS_SET(vm->vmRuntimeStateListener.idleTuningFlags, J9_IDLE_TUNING_GC_ON_IDLE | J9_IDLE_TUNING_COMPACT_ON_IDLE)) {
 				BOOLEAN idleGCTuningSupported = FALSE;
-#if (defined(LINUX) && (defined(J9HAMMER) || defined(J9X86))) || defined(J9ZOS39064)
+#if (defined(LINUX) && (defined(J9HAMMER) || defined(J9X86) || defined(S390) || defined(LINUXPPC64) || (defined(LINUXPPC64) && defined(J9VM_ENV_LITTLE_ENDIAN)))) || defined(J9ZOS39064)
```
Comment (pshipton) [inline runtime/vm/jvminit.c]: `(defined(LINUXPPC64) && defined(J9VM_ENV_LITTLE_ENDIAN))` is redundant considering `defined(LINUXPPC64)` is also added

**Comment by pshipton** [comment]
jenkins test all plinux,zlinux

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK8-linux_390-64_cmprssptrs-OpenJ9/109/
https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK8-linux_ppc-64_cmprssptrs_le-OpenJ9/213/
https://ci.eclipse.org/openj9/job/PullRequest-Extended-JDK8-linux_390-64_cmprssptrs-OpenJ9/38/
https://ci.eclipse.org/openj9/job/PullRequest-Extended-JDK8-linux_ppc-64_cmprssptrs_le-OpenJ9/56/

https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK9-linux_390-64_cmprssptrs-OpenJ9/108/
https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK9-linux_ppc-64_cmprssptrs_le-OpenJ9/231/
https://ci.eclipse.org/openj9/job/PullRequest-Extended-JDK9-linux_390-64_cmprssptrs-OpenJ9/37/
https://ci.eclipse.org/openj9/job/PullRequest-Extended-JDK9-linux_ppc-64_cmprssptrs_le-OpenJ9/60/

https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK10-linux_390-64_cmprssptrs-OpenJ9/20/
https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK10-linux_ppc-64_cmprssptrs_le-OpenJ9/39/
https://ci.eclipse.org/openj9/job/PullRequest-Extended-JDK10-linux_390-64_cmprssptrs-OpenJ9/2/
https://ci.eclipse.org/openj9/job/PullRequest-Extended-JDK10-linux_ppc-64_cmprssptrs_le-OpenJ9/9/

File: runtime/vm/jvminit.c (hunk 1 above)
Comment (Param-S) [reply runtime/vm/jvminit.c]: Yes, corrected it.

File: runtime/vm/jvminit.c [hunk 2]
```diff
@@ -1952,7 +1952,7 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 			 */
 			if (J9_ARE_ANY_BITS_SET(vm->vmRuntimeStateListener.idleTuningFlags, J9_IDLE_TUNING_GC_ON_IDLE | J9_IDLE_TUNING_COMPACT_ON_IDLE)) {
 				BOOLEAN idleGCTuningSupported = FALSE;
-#if (defined(LINUX) && (defined(J9HAMMER) || defined(J9X86))) || defined(J9ZOS39064)
+#if (defined(LINUX) && (defined(J9HAMMER) || defined(J9X86) || defined(S390) || defined(PPC64))) || defined(J9ZOS39064)
```
Comment (pshipton) [inline runtime/vm/jvminit.c]: S390 is for 32-bit as well as 64-bit. Plus I think its only defined for jit code.
PPC64 also includes AIX.

**Comment by pshipton** [comment]
jenkins compile plinux,zlinux jdk8

File: runtime/vm/jvminit.c [hunk 3]
```diff
@@ -1952,7 +1952,7 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 			 */
 			if (J9_ARE_ANY_BITS_SET(vm->vmRuntimeStateListener.idleTuningFlags, J9_IDLE_TUNING_GC_ON_IDLE | J9_IDLE_TUNING_COMPACT_ON_IDLE)) {
 				BOOLEAN idleGCTuningSupported = FALSE;
-#if (defined(LINUX) && (defined(J9HAMMER) || defined(J9X86))) || defined(J9ZOS39064)
+#if (defined(LINUX) && (defined(J9HAMMER) || defined(J9X86) || defined(S39064) || defined(PPC64))) || defined(J9ZOS39064)
```
Comment (charliegracie) [inline runtime/vm/jvminit.c]: I am curious why this is not wrapped via the OMR_GC_IDLE_HEAP_MANAGER flag instead of this complicated list of SPECs?

**Pull request merged**